src/timeout_func.rs

```diff
@@ -0,0 +1,32 @@
+// This file is part of rgtk.
+//
+// rgtk is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// rgtk is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with rgtk.  If not, see <http://www.gnu.org/licenses/>.
+
+pub mod timeout {
+    use ffi;
+
+    pub fn add<T>(interval: u32, func: fn(&T) -> i32, data: &T) -> u32 {
+        let tmp = data as *const T;
+        let tmp_f = func as ffi::gpointer;
```

Comment:
`func` isn't `extern "C"` so you're asking for trouble here. There's a reason `signal.rs` in `gtk` uses trampolines.